### PR TITLE
Add Dockerfile and workflow for course discovery

### PR DESCRIPTION
**Pull request by mfarhan943**
**Description**
This PR adds Dockerfile and workflow to push the image for course discovery in the repo.

**Issue Link**
https://github.com/edx/public-dockerfiles/issues/4

### CODE REVIEW

File: .github/workflows/push-course-discovery-image.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Send failure notification
+        if: failure()
+        uses: dawidd6/action-send-mail@v3
+        with:
+          server_address: email-smtp.us-east-1.amazonaws.com
+          server_port: 465
+          username: ${{secrets.edx_smtp_username}}
+          password: ${{secrets.edx_smtp_password}}
+          subject: Push Image to docker.io/edxops failed in Course Discovery
+          to: team-cosmonauts@edx.org
```
Comment (UsamaSadiq) [inline .github/workflows/push-course-discovery-image.yaml:55]: Owning team for course-discovery is `phoenix`. We should ask the team if they have an email address available for tagging. If not then, let's ask them which email address they would like to be notified on for the failure notification.

**Review by UsamaSadiq — CHANGES_REQUESTED** [review]

**Comment by UsamaSadiq** [comment]
@mfarhan943 since you have now write access to the repository, let's close this PR and create a PR by pushing the changes directly. 
Then we can also add `pull_request` header and test the build on the PR itself.

File: .github/workflows/push-course-discovery-image.yaml (hunk 1 above)
Comment (mfarhan943) [reply .github/workflows/push-course-discovery-image.yaml:55]: I have updated the email in new PR.

**Comment by mfarhan943** [comment]
Closing this PR as a new PR with the direct branch has been opened.
https://github.com/edx/public-dockerfiles/pull/41

**Pull request closed without merging**